script_test.go

```diff
@@ -1768,6 +1769,57 @@ func TestWithStdout_SetsSpecifiedWriterAsStdout(t *testing.T) {
 	}
 }
 
+func TestWithEnv_SetEmptyEnvironment(t *testing.T) {
+	t.Parallel()
+	buf := new(bytes.Buffer)
+	env := []string{}
+
+	_, err := script.NewPipe().WithStdout(buf).WithEnv(env).Exec("printenv").Stdout()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(buf.String()) != 0 {
+		got := strings.Split(strings.TrimRight(buf.String(), "\n"), "\n")
+		t.Errorf("expected 0 environment variables, got %d %v", len(got), got)
```

Comment:
I'm not sure the string-munging is all that helpful here, and clutters up the test a bit, making it hard to see what's going on. The point you're making is that the string should be empty. If it's not, we want to know what's in it. That's all.

```suggestion
		if output != "" {
		        t.Errorf("want empty environment, got %q", output)
		 }
```